packages/SwingSet/test/virtualObjects/test-representatives.js

```diff
@@ -279,60 +279,162 @@ test('exercise cache', async t => {
   t.deepEqual(log.shift(), ['set', thingID(2), thingVal('thing2 updated')]);
+  t.deepEqual(log, []);
+
   await read(T5, 'thing5'); // reanimate t5, evict t1 - [t5 t6 t7 t8]
   t.deepEqual(log.shift(), ['get', thingID(5), thingVal('thing5')]);
   t.deepEqual(log.shift(), ['set', thingID(1), thingVal('thing1 updated')]);
+  t.deepEqual(log, []);
+
   await read(T4, 'thing4'); // reanimate t4, evict t8 - [t4 t5 t6 t7]
   t.deepEqual(log.shift(), ['get', thingID(4), thingVal('thing4')]);
   t.deepEqual(log.shift(), ['set', thingID(8), thingVal('thing8')]);
+  t.deepEqual(log, []);
+
   await read(T3, 'thing3'); // reanimate t3, evict t7 - [t3 t4 t5 t6]
   t.deepEqual(log.shift(), ['get', thingID(3), thingVal('thing3')]);
+  t.deepEqual(log.shift(), ['get', `${thingID(7)}.refCount`, '1 0']);
   t.deepEqual(log.shift(), ['set', thingID(7), thingVal('thing7')]);
+  t.deepEqual(log, []);
 
   await read(T2, 'thing2 updated'); // reanimate t2, evict t6 - [t2 t3 t4 t5]
   t.deepEqual(log.shift(), ['get', thingID(2), thingVal('thing2 updated')]);
+  t.deepEqual(log.shift(), ['get', `${thingID(6)}.refCount`, '1 0']);
+  t.deepEqual(log, []);
+
   await readHeld('thing1 updated'); // reanimate t1, evict t5 - [t1 t2 t3 t4]
   t.deepEqual(log.shift(), ['get', thingID(1), thingVal('thing1 updated')]);
+  t.deepEqual(log.shift(), ['get', `${thingID(5)}.refCount`, '1 0']);
+  t.deepEqual(log, []);
 
   await forgetHeld(); // cache unchanged - [t1 t2 t3 t4]
+  t.deepEqual(log.shift(), ['get', `${thingID(1)}.refCount`, '1 0']);
+  t.deepEqual(log, []);
+
   await hold(T8); // cache unchanged - [t1 t2 t3 t4]
+  t.deepEqual(log.shift(), ['get', `${thingID(4)}.refCount`, '1 0']);
   t.deepEqual(log, []);
+
   await read(T7, 'thing7'); // reanimate t7, evict t4 - [t7 t1 t2 t3]
   t.deepEqual(log.shift(), ['get', thingID(7), thingVal('thing7')]);
+  t.deepEqual(log.shift(), ['get', `${thingID(3)}.refCount`, '1 0']);
+  t.deepEqual(log, []);
+
   await writeHeld('thing8 updated'); // reanimate t8, evict t3 - [t8 t7 t1 t2]
   t.deepEqual(log.shift(), ['get', thingID(8), thingVal('thing8')]);
   t.deepEqual(log, []);
 });
+
+test('virtual object gc', async t => {
+  const config = {
+    bootstrap: 'bootstrap',
+    defaultManagerType: 'xs-worker',
+    vats: {
+      bob: {
+        sourceSpec: path.resolve(__dirname, 'vat-vom-gc-bob.js'),
+        creationOptions: {
+          virtualObjectCacheSize: 3,
+        },
+      },
+      bootstrap: {
+        sourceSpec: path.resolve(__dirname, 'vat-vom-gc-bootstrap.js'),
+      },
+    },
+  };
+
+  const hostStorage = provideHostStorage();
+
+  const c = await buildVatController(config, [], { hostStorage });
+  c.pinVatRoot('bootstrap');
+
+  await c.run();
+  t.deepEqual(
+    c.kpResolution(c.bootstrapResult),
+    capargs({ '@qclass': 'undefined' }),
+  );
+  const remainingVOs = {};
+  for (const key of hostStorage.kvStore.getKeys('v1.vs.', 'v1.vs/')) {
```

Comment:
It might be good to get `v1` by calling `c.vatNameToID(`bob`) rather than hard-coding it.

Reply:
Meh.  The ordering is determined by the config file. I'm not sure writing more code adds anything.